### ISSUE TITLE
Add JCasC to the Jenkinsfile Runner Vanilla Package

### ISSUE THREAD

**Issue by oleg-nenashev** [jenkinsci/jenkinsfile-runner#197]
https://github.com/jenkinsci/configuration-as-code-plugin is the base way to configure Jenkins modern instances, and it works well with Jenkinsfile Runner.

Acceptance criteria:

* The plugin dependency is added to the bundle. Version can be taken from BOM
* JCasC directory is exposed as a volume so that users can append their configs
* There is a documentation section JCasC with Jenkinsfile Runner
* There is a demo created which demonstrates how to use the plugin inside Jenkinsfile Runner

FYI @casz @timja

**Comment by literalplus** [comment]
I created a small PoC in a branch: https://github.com/jenkinsci/jenkinsfile-runner/compare/master...xxyy:197/jasc-support

Output (note the system-wide env var set by CasC): https://gist.github.com/xxyy/c77783bd942e3ac76ffec974003b82ad

Is this about what you were thinking with this? If it is, I'll add the docs as well & submit a PR for this. The demo from the output is in the branch as well, let me know if it needs expanding.

To do:
 - [x] the directory support of casc doesn't see to work. It reads the directory as space-separated list of the file names (?? - maybe I'm using it wrong)
 - [x] warning printed when not providing a jenkins.yaml because the default is an empty file
 - [x] docs

BTW: I noticed that running the build on JDK 13 fails because javax.annotations is missing in the bootstrap module (no longer bundled with the JDK). Should I open a PR for that as well? (I know JDK 13 is not supported by Jenkins, but this should happen with JDK 11 as well)
